### DISCUSSION TITLE
useForm: update initialValues

### DISCUSSION THREAD

**Discussion by testerez** [mantinedev/mantine#2576] [Feature requests]
It would be great if I could just update initial values and get actual values updated for not dirty fields.

This would help when initial values require data fetching. Or when some initial values depend on selected values in the form.

**Comment by rtivital** [comment]
You can call `form.resetDirty` to update values that are used for field status comparison.

**Reply by testerez** [reply]
I don't think `resetDirty` would work for what I'm trying to achieve. Though, I was able to get the behaviour I wanted with something like this:
```ts
Object.entries(data.values).forEach(([key, value]) => {
  if (!form.isDirty(key)) {
    form.setFieldValue(key, value)
  }
})
```
